torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -428,105 +484,124 @@ void ProcessGroupAgent::enqueueSend(SendWork work) {
       std::move(work)));
 }
 
-void ProcessGroupAgent::enqueueRecv(RecvWork work) {
-  threadPool_.run(std::bind(
-      [&](RecvWork& work) {
-        torch::Tensor& payload = work.payload_;
-        auto data = wireDeserialize(payload.storage().data(), payload.numel());
-        Message message(
-            std::move(data.first),
-            std::move(data.second),
-            work.type_,
-            work.id_);
-        if (message.isRequest()) {
-          ++serverActiveCalls_;
-          std::shared_ptr<FutureMessage> futureResponse;
-          try {
-            futureResponse = cb_->operator()(message);
-          } catch (const std::exception& e) {
-            futureResponse = std::make_shared<FutureMessage>();
-            futureResponse->setError(e.what());
-          }
-          if (futureResponse->completed()) {
+int ProcessGroupAgent::handleRecv(RecvWork& work) {
```

Comment:
nit: (fix can come in the next PR is necessary) any reason for returning 0/1 instead of false/true here? Do we plan to return other values in the future?

Reply:
This is true, it should return a bool. I will update it in the next diff.